crates/matrix-sdk-crypto/src/olm/group_sessions/outbound.rs

```diff
@@ -356,30 +358,37 @@ impl OutboundGroupSession {
     ///
     /// # Arguments
     ///
-    /// * `content` - The plaintext content of the message that should be
-    /// encrypted in raw json [`Value`] form.
-    ///
     /// * `event_type` - The plaintext type of the event, the outer type of the
     /// event will become `m.room.encrypted`.
     ///
+    /// * `content` - The plaintext content of the message that should be
+    /// encrypted in raw JSON form.
+    ///
     /// # Panics
     ///
     /// Panics if the content can't be serialized.
     pub async fn encrypt(
         &self,
-        content: Value,
         event_type: &str,
+        content: &Raw<AnyMessageLikeEventContent>,
     ) -> Raw<RoomEncryptedEventContent> {
-        let json_content = json!({
-            "content": content,
-            "room_id": &*self.room_id,
-            "type": event_type,
-        });
+        #[derive(Serialize)]
+        struct Payload<'a> {
+            #[serde(rename = "type")]
```

Comment:
We probably have a Ruma type for this nowadays, but it's unlikely going to be better than this borrowed type.

Reply:
We have [`DecryptedMegolmV1Event`](https://docs.rs/ruma/latest/ruma/events/struct.DecryptedMegolmV1Event.html) but it takes the event type from content (which requires the content to implement [`MessageLikeEventContent`](https://docs.rs/ruma/latest/ruma/events/trait.MessageLikeEventContent.html), which doesn't work for raw events. And even if that wasn't the case, the borrowing is also nice in a potentially hot path like encryption.